velox/exec/Driver.h

```diff
@@ -470,6 +490,34 @@ class Driver : public std::enable_shared_from_this<Driver> {
   }
 
  private:
+  // Ensures that the thread is removed from its Task's thread count on exit.
```

Comment:
Let's keep CancelGuard in cc? Any reason to move it here as subclass? Thanks!

Reply:
It is needed in the signature of new `blockDriver` method.  This is private so we could put a forward declaration in class and keep the actual definition in cpp file.